src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -50,6 +50,16 @@ internal struct ReliabilitySection
             internal static void Assert(string message)
             {
             }
+
+            [Conditional("NETFRAMEWORK")]
```

Comment:
I'm a bit confused why something related to netfx is being added to the netcore project...

Reply:
The .NET Framework code's `WaitForPendingOpen` method uses the ReliabilitySection struct, and calls Start and Stop. The ReliabilitySection struct already exists in both [netcore](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs#L44) and [netfx](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs#L89), so I added two methods to it. The `[Conditional]` attribute on the method means that its call will always be compiled out.

ReliabilitySection in general will always no-op in .NET Core - ReliabilitySection sets up Constrained Execution Regions, which only do anything in .NET Framework.

Reply:
Ok - I've got merging TdsParser on my stack ... and it should be "fun". No problem with this for now.